components/playlist/playlist_download_request_manager.cc

```diff
@@ -159,21 +146,28 @@ void PlaylistDownloadRequestManager::DidFinishLoad(
   if (in_progress_urls_count_ == 0 || callback_for_current_request_.is_null())
     return;
 
-  DCHECK(web_contents_->GetMainFrame());
+  GetMedia(web_contents_.get());
+}
 
-  web_contents_->GetMainFrame()->ExecuteJavaScriptInIsolatedWorld(
+void PlaylistDownloadRequestManager::GetMedia(content::WebContents* contents) {
+  DCHECK(contents && contents->GetMainFrame());
+
+  contents->GetMainFrame()->ExecuteJavaScriptInIsolatedWorld(
       base::UTF8ToUTF16(media_detector_script_),
       base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
-                     weak_factory_.GetWeakPtr()),
+                     weak_factory_.GetWeakPtr(), contents->GetWeakPtr()),
       g_playlist_javascript_world_id);
 }
 
-void PlaylistDownloadRequestManager::OnGetMedia(base::Value value) {
-  auto print_value = [](const base::Value& value) {
-    std::string out;
-    base::JSONWriter::Write(value, &out);
-    LOG(ERROR) << value;
-  };
+void PlaylistDownloadRequestManager::OnGetMedia(
+    base::WeakPtr<content::WebContents> contents,
+    base::Value value) {
+  if (!contents)
+    return;
+
+  DCHECK(!callback_for_current_request_.is_null()) << " callback already ran";
+  auto callback = std::move(callback_for_current_request_);
+  DCHECK(callback_for_current_request_.is_null());
```

Comment:
I'm pretty sure `std::move()` will work, we don't need to debug flying neutrinos 😄. let's remove this line.

Reply:
Yes 😆. Removed.